tests/unit/notification/NotificationConfigManager.js

```diff
@@ -0,0 +1,178 @@
+        assert.throws(() => new NotificationConfigManager({
+            zkClient: null,
+            logger: null,
+        }));
+        const manager = new NotificationConfigManager(params);
+        assert(manager instanceof NotificationConfigManager);
+        async.series([
+            next => managerInit(manager, next),
+            next => checkCount(manager, next),
+        ], done);
+    });
+
+    it('should get bucket notification configuration', () => {
+        const manager = new NotificationConfigManager(params);
+        managerInit(manager, () => {
+            const bucket = 'bucket1';
+            const result = manager.getConfig(bucket);
+            assert.strictEqual(result.bucket, bucket);
+        });
+    });
+
+    it('should return undefined for an invalid bucket', () => {
+        const manager = new NotificationConfigManager(params);
+        managerInit(manager, () => {
+            const bucket = 'bucket100';
+            const result = manager.getConfig(bucket);
+            assert.strictEqual(result, undefined);
+        });
+    });
+
+    it('should add bucket notification configuration', done => {
+        const manager = new NotificationConfigManager(params);
+        managerInit(manager, () => {
+            const bucket = 'bucket100';
+            const config = getTestConfigValue(bucket);
+            const result = manager.setConfig(bucket, config);
+            assert(result);
+            setTimeout(() => {
+                checkCount(manager, done);
+            }, timeoutMs);
+        });
+    });
+
+    it('should update bucket notification configuration', done => {
+        const manager = new NotificationConfigManager(params);
+        managerInit(manager, () => {
+            const bucket = 'bucket1';
+            const config = getTestConfigValue(`${bucket}-updated`);
+            const result = manager.setConfig(bucket, config);
+            assert(result);
+            setTimeout(() => {
+                const updatedConfig = manager.getConfig(bucket);
+                assert.strictEqual(updatedConfig.bucket, `${bucket}-updated`);
+                checkCount(manager, done);
+            }, timeoutMs);
+        });
+    });
+
+    it('should remove bucket notification configuration', done => {
+        const manager = new NotificationConfigManager(params);
+        managerInit(manager, () => {
+            const bucket = 'bucket1';
+            const result = manager.removeConfig(bucket);
+            assert(result);
+            setTimeout(() => {
+                checkCount(manager, done);
+            }, timeoutMs);
+        });
+    });
+
+    it('config count should be zero when all configs are removed', done => {
+        const manager = new NotificationConfigManager(params);
+        async.series([
+            next => managerInit(manager, next),
+            next => deleteTestConfigs(zkClient, next),
+            next => setTimeout(next, 100),
```

Comment:
I assume you could use `timeoutMs` since you defined it

Reply:
:facepalm: my copy paste skills are not good enough.